tests/explainers/test_tree.py

```diff
@@ -269,6 +289,24 @@ def test_lightgbm_multiclass():
     # ensure plot works for first class
     shap.dependence_plot(0, shap_values[0], X, show=False)
 
+def test_lightgbm_ranking():
+    try:
+        import lightgbm
+    except:
+        print("Skipping test_lightgbm_ranking!")
+        return
+    import shap
+
+    # train lightgbm ranker model
+    x_train, y_train, x_test, y_test, q_train, q_test = shap.datasets.rank()
+    model = lightgbm.LGBMRanker()
+    model.fit(x_train, y_train, group=q_train, eval_set=[(x_test, y_test)],
+              eval_group=[q_test], eval_at=[1, 3], early_stopping_rounds=5, verbose=False,
+              callbacks=[lightgbm.reset_parameter(learning_rate=lambda x: 0.95 ** x * 0.1)])
+
+    # explain the model's predictions using SHAP values
+    shap.TreeExplainer(model).shap_values(x_test)
```

Comment:
Same as above checking that the values sum to the margin prediction of the model would be good (plus expected_value of course).

Reply:
good idea, done!